packages/models/src/lib/budgets.ts

```diff
@@ -0,0 +1,50 @@
+import { z } from 'zod';
+
+/**
+ * Define Zod schema for the Budget type
+ *
+ * @example
+ *
+ * // Type assertion for valid budget data
+ * const validBudget = {
+ *   ref: 'eslint#max-lines',
+ *   minScore: 0.8,
+ *   maxWarnings: 5,
+ *   maxValue: 1000,
+ * } as z.infer<typeof budgetSchema>;
+ *
+ * // Validate the data against the schema
+ * const validationResult = budgetSchema.safeParse(validBudget);
+ *
+ * if (validationResult.success) {
+ *   console.log('Valid budget:', validationResult.data);
+ * } else {
+ *   console.error('Invalid budget:', validationResult.error);
+ * }
+ *
+ **/
+export const budgetSchema = z.object({
+  ref: z.string({
+    description:
+      "reference to audit ('eslint#max-lines') or category ('categories:performance')",
+  }),
+  minScore: z
+    .number({
+      description: 'fail assertion if score too low',
+    })
+    .min(0)
+    .max(1)
+    .optional(),
+  maxWarnings: z
+    .number({
+      description: 'fail assertion if too many warnings',
+    })
+    .optional(),
```

Comment:
The `maxWarnings` doesn't quite fit with the issue models we have now :thinking: Instead of just a list warnings, we now have a list issues which have a `severity` of `info`, `warning` or `error` (could be extended).

I guess the best way would be to have a configurable rule with a minimum severity level. E.g. `maxIssues: { count: 0, severity: 'warning' }` would mean the budget fails when there are any `warning` or `error` issues, but passes when there are only `info` issues.

But the budgets are nice-to-have for me now anyway, so I'd also be OK with not including them in the codebase yet.

Reply:
Will remove them!

Reply:
done